test/common/http/path_utility_test.cc

```diff
@@ -69,6 +87,32 @@ TEST_F(PathUtilityTest, NormalizeValidPaths) {
   }
 }
 
+// Already normalized path don't change.
+TEST(PathTransformerTest, NormalizeValidPaths) {
+
+  const std::vector<std::pair<std::string, std::string>> non_normal_pairs{
+      {"/a/b/../c", "/a/c"},        // parent dir
+      {"/a/b/./c", "/a/b/c"},       // current dir
+      {"a/b/../c", "/a/c"},         // non / start
+      {"/a/b/../../../../c", "/c"}, // out number parent
+      {"/a/..\\c", "/c"},           // "..\\" canonicalization
+      {"/%c0%af", "/%c0%af"},       // 2 bytes unicode reserved characters
+      {"/%5c%25", "/%5c%25"},       // reserved characters
+      {"/a/b/%2E%2E/c", "/a/c"}     // %2E escape
```

Comment:
Could you also test paths like: `"/a/b/%2E./c"` and `"/a/b/.%2E/c"`? Also a path with a single dot fragment represent by a %-encoding, like `"/a/b/%2E/c"`? There's also a case of a `..` as a first path fragment: `"/../a/b"`, ditto for a `.`.

Reply:
Thanks, will do.